packages/angular_devkit/build_angular/src/tools/esbuild/stylesheets/bundle-options.ts

```diff
@@ -152,14 +152,14 @@ export async function bundleComponentStylesheet(
   if (!result.errors) {
     for (const outputFile of result.outputFiles) {
       const filename = path.basename(outputFile.path);
-      if (filename.endsWith('.css')) {
+      if (outputFile.type === BuildOutputFileType.Media) {
```

Comment:
Can we add an else block with an assert fail to cover any non CSS/media files just in case?  This should not happen and if it does it probably means there is a bug somewhere.